### PR TITLE
Rename a number of types and modules.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

In general we should try to be specific about what we're referring to.
"Structured headers" is not really an accurate description of what this
module works with: "Structured Header Field Values", or "Structured
Field Value" is usually better. To that end this change adjusts our
naming to be closer to what we want. It also changes our module names to
prioritise the use of the Codable-based one.

Modifications:

- Renamed CodableStructuredHeaders to StructuredFieldValues.
- Renamed StructuredHeaders to RawStructuredFieldValues.
- Renamed a number of types from Field to FieldValue.
- Renamed a number of types from Header to Field or FieldValue.
- Renamed some functions along the above lines.

Result:

Better and more consistent terminology will be used.

### CODE REVIEW

File: scripts/soundness.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ set -eu
 here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 function replace_acceptable_years() {
     # this needs to replace all acceptable forms with 'YEARS'
-    sed -e 's/2020/YEARS/'
+    sed  -e 's/2020-2021/YEARS/' -e 's/202[01]/YEARS/'
```
Comment (PeterAdams-A) [inline scripts/soundness.sh:20]: Probably should have done this as a separate PR but never mind.

File: README.md [hunk 2]
```diff
@@ -41,66 +41,66 @@ In most cases users should prefer to use `CodableStructuredHeaders` unless they
...
+5. Strings. `StructuredFieldValues` models these as Swift's `String` type.
+6. Binary data. `StructuredFieldValues` models this as Foundation's `Data` type.
 
-For any Structured Header Field Item, the item may either be represented directly by the appropriate type, or by a Swift struct with two properties: `item` and `parameters`. This latter mode is how parameters on a given item may be captured.
+For any Structured Header Field Value Item, the item may either be represented directly by the appropriate type, or by a Swift struct with two properties: `item` and `parameters`. This latter mode is how parameters on a given item may be captured.
 
-The top-level structured header field must identify what kind of header field it corresponds to: `.item`, `.list`, or `.dictionary`. This is inherent in the type of the field and will be specified in the relevant field specification.
+The top-level structured header field value must identify what kind of header field it corresponds to: `.item`, `.list`, or `.dictionary`. This is inherent in the type of the field and will be specified in the relevant field specification.
```
Comment (PeterAdams-A) [inline README.md:77]: Not sure why you've made the capitalisation different between this line and the one above but it was like that before.

**Review by PeterAdams-A — APPROVED** [review]
Seems reasonable.

File: README.md [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
 
-Provides parsing and serialization facilities for structured header fields, as well as implementations of `Encoder` and `Decoder` to allow using `Codable` data types as the payloads of HTTP structured header fields.
+Provides parsing and serialization facilities for structured header field values, as well as implementations of `Encoder` and `Decoder` to allow using `Codable` data types as the payloads of HTTP structured header fields.
 
-## About Structured Header Fields
+## About Structured Header Field Values
 
-HTTP Structured Header fields are a HTTP extension recorded in [RFC 8941](https://www.rfc-editor.org/rfc/rfc8941.html) . They provide a set of data types and algorithms for handling HTTP header field values in a consistent way, allowing a single parser and serializer to handle a wide range of header field values.
+HTTP Structured Header Field Values are a HTTP extension recorded in [RFC 8941](https://www.rfc-editor.org/rfc/rfc8941.html) . They provide a set of data types and algorithms for handling HTTP header field values in a consistent way, allowing a single parser and serializer to handle a wide range of header field values.
```
Comment (glbrntt) [inline README.md:9]:
```suggestion
HTTP Structured Header Field Values are a HTTP extension recorded in [RFC 8941](https://www.rfc-editor.org/rfc/rfc8941.html). They provide a set of data types and algorithms for handling HTTP header field values in a consistent way, allowing a single parser and serializer to handle a wide range of header field values.
```

File: README.md [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
 
-In most cases users should prefer to use `CodableStructuredHeaders` unless they know they need the performance advantages of  `StructuredHeaders`. The experience will be much better.
+In most cases users should prefer to use `StructuredFieldValues` unless they know they need the performance advantages of `RawStructuredFieldValues`. The experience will be much better.
 
-## Working with Structured Header Fields
+## Working with Structured Header Field Values
 
-`swift-http-structured-headers` has a simply, easy-to-use high-level API for working with structured header fields. To begin with, let's consider the [HTTP Client Hints specification](https://www.rfc-editor.org/rfc/rfc8942.html). This defines the following new header field:
+`swift-http-structured-headers` has a simply, easy-to-use high-level API for working with structured header field values. To begin with, let's consider the [HTTP Client Hints specification](https://www.rfc-editor.org/rfc/rfc8942.html). This defines the following new header field:
```
Comment (glbrntt) [inline README.md:27]:
```suggestion
`swift-http-structured-headers` has a simple, easy-to-use high-level API for working with structured header field values. To begin with, let's consider the [HTTP Client Hints specification](https://www.rfc-editor.org/rfc/rfc8942.html). This defines the following new header field:
```

**Review by glbrntt — APPROVED** [review]
Left suggestions for a few existing typos. Looks good otherwise. 

Did you consider adding availability annotations for renaming? I suspect it might be more effort than it's worth though.

**Review by fabianfett — APPROVED** [review]
I like the new names
